module/userProjects/helper.js

```diff
@@ -1742,173 +1844,246 @@ module.exports = class UserProjectsHelper {
 								taskReport[task.status] += 1
 							}
 						} else {
-							//if task is deleted it is not counted in total.
 							taskReport.total = taskReport.total - 1
 						}
 					})
+					project['taskReport'] = taskReport
 
-					createProject['taskReport'] = taskReport
-				}
-
-				let booleanData = this.booleanData(schemas['projects'].schema)
-				let mongooseIdData = this.mongooseIdData(schemas['projects'].schema)
+					// Attach programInformation to each project
+					project['programInformation'] = programInformation
+					project['programId'] = programInformation._id
+					project['programExternalId'] = programInformation.externalId
+					project['isAPrivateProgram'] = programInformation.isAPrivateProgram
 
-				Object.keys(data).forEach((updateData) => {
-					if (!createProject[updateData] && projectsModel.includes(updateData)) {
-						if (booleanData.includes(updateData)) {
-							createProject[updateData] = UTILS.convertStringToBoolean(data[updateData])
-						} else if (mongooseIdData.includes(updateData)) {
-							createProject[updateData] = ObjectId(data[updateData])
-						} else {
-							createProject[updateData] = data[updateData]
-						}
+					// Attach app information to each project
+					project['appInformation'] = {}
+					if (appName !== '') {
+						project['appInformation']['appName'] = appName
 					}
-				})
 
-				createProject['appInformation'] = {}
-				if (appName !== '') {
-					createProject['appInformation']['appName'] = appName
-				}
+					if (appVersion !== '') {
+						project['appInformation']['appVersion'] = appVersion
+					}
 
-				if (appVersion !== '') {
-					createProject['appInformation']['appVersion'] = appVersion
-				}
+					project['lastDownloadedAt'] = new Date()
 
-				createProject['lastDownloadedAt'] = new Date()
+					if (data.profileInformation) {
+						project.userRoleInformation = project.profileInformation
+					}
 
-				if (data.profileInformation) {
-					createProject.userRoleInformation = data.profileInformation
-				}
+					project.status = UTILS.convertProjectStatus(project.status)
 
-				createProject.status = UTILS.convertProjectStatus(data.status)
-				let userProject = await projectQueries.createProject(createProject)
+					// Fetch and attach category information to each project
+					project['category'] = project['category']
+						? project['category']
+						: [process.env.DEFAULT_PROJECT_CATEGORY]
+					const categories = await projectCategoriesQueries.categoryDocuments(
+						{
+							externalId: { $in: project.category },
+						},
+						['_id', 'name', 'externalId', 'evidences']
+					)
+					project['categories'] = categories
+					project['userId'] = project['createdBy'] = project['updatedBy'] = userId
+
+					// Attach solution information to each project
+					for (const solution of programAndMultipleSolutionsData.data.solutions) {
+						if (solution.name == project.title) {
+							project['solutionInformation'] = {
+								_id: solution._id,
```

Comment:
@Prajwal17Tunerlabs can introduce a new key called reflectionEnabled. this can be set in env. default value false, but for MI this can be true from .env. andsolution creation data doesn't have reflectionEnabled data take it from env and add to solution when we create it. And store it in solutionInformation of projects env key can be ENABLE_REFLECTION

Reply:
I have made the changes as mentioned above